enterprise/models.py

```diff
@@ -2511,12 +2513,41 @@ class DefaultEnterpriseEnrollmentIntention(TimeStampedModel, SoftDeletableModel)
     )
     history = HistoricalRecords()
 
+    def get_cached_content_metadata(self):
+        """
+        Retrieves the cached content metadata for the instance content_key
+        Determines catalog from the catalogs associated to the enterprise customer
+        """
+        try:
+            content_metadata = get_and_cache_catalog_content_metadata(
+                enterprise_customer=self.enterprise_customer,
+                content_keys=[self.content_key],
+            )
+            return content_metadata
+        except Exception as exc:  # pylint: disable=broad-except
+            LOGGER.exception(
+                "Unable to retrieve content_metadata for customer"
+                "enterprise_customer:{}, "
+                "content_key:{}"
+                .format(self.enterprise_customer, self.content_key),
+                exc_info=exc,
+            )
+            return None
+
     @cached_property
     def current_course_run(self):  # pragma: no cover
         """
         Metadata describing the current course run for this default enrollment intention.
         """
-        return {}
+        content_metadata_items = self.get_cached_content_metadata()
+        if not content_metadata_items:
+            return {}
+
+        content_metadata_item = content_metadata_items[0]
+        if self.content_type == 'course':
+            return get_advertised_course_run(content_metadata_item)
```

Comment:
[inform] `get_advertised_course_run` returns `None` if there is no `advertised_course_run_uuid` (e.g., `null`). We may want to ensure consistency in the return value (e.g., `if None: return {}`).